src/webui/api/torrentscontroller.cpp

```diff
@@ -849,16 +986,47 @@ void TorrentsController::addAction()
         }
     };
 
+
     bool partialSuccess = false;
-    for (QString url : asConst(urls.split(u'\n')))
+    for (QString url : urls)
     {
         url = url.trimmed();
         if (!url.isEmpty())
         {
+            auto iter = m_torrentSource.find(url);
```

Comment:
If `iter` is not going to be mutated then please replace it with `m_torrentSource.constFind(url);`
and `m_torrentSource.end())` with `m_torrentSource.constEnd())`